File: app/src/main/java/com/ubiqsmart/ui/widgets/charts/WeekXFormatter.java
package com.ubiqsmart.ui.widgets.charts;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.IAxisValueFormatter;

import java.util.*;

public class WeekXFormatter implements IAxisValueFormatter {

  private GregorianCalendar c = new GregorianCalendar();

  @Override public String getFormattedValue(float value, AxisBase axis) {
    c.setTimeInMillis(((long) (value)) * 1000);
    return c.get(Calendar.DAY_OF_MONTH) + ". ";
  }
}

File: app/src/main/java/com/ubiqsmart/ui/widgets/charts/WeekXFormatter.kt
package com.ubiqsmart.ui.widgets.charts

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.IAxisValueFormatter

import java.util.*

class WeekXFormatter : IAxisValueFormatter {

  private val c = GregorianCalendar()

  override fun getFormattedValue(value: Float, axis: AxisBase): String {
    c.timeInMillis = value.toLong() * 1000
    return c.get(Calendar.DAY_OF_MONTH).toString() + ". "
  }
}
